packages/nextjs/src/config/webpack.ts

```diff
@@ -142,6 +142,8 @@ export function constructWebpackConfigFunction(
       // with the `--ignore-scripts` option, this will be blocked and the missing binary will cause an error when users
       // try to build their apps.)
       ensureCLIBinaryExists() &&
+      // TODO: We might want to add detection for other non-prod environments
+      !isVercelNonProdEnv() &&
```

Comment:
This name makes my brain go wild. Can we do this instead?

```diff
- !isVercelNonProdEnv() &&
+ isVercelProdEnv() &&
```

And then later update the function to represent it, like so:
```javascript
/** Detect whether this build is part of a prod deployment on vercel, in order to decide whether to run the webpack
 * plugin */
function isVercelProdEnv(): boolean {
  return process.env.VERCEL_ENV === 'production';
}
```

Reply:
I started that way, but switched it because we also want to enable the plugin off of Vercel. If we do a `isVercelProd` check where I currently have it, we'll cut out non-prod Vercel (good) but also non-Vercel-everything (bad). Essentially, what we want is a deny-list, not an allow-list.

Regardless, in the log run, this will become a more generic function (`shouldDisablePlugin` or something), so I don't think the naming for the stop-gap solution is something we have to worry too much about.